libs/SalesforceSDK/src/com/salesforce/androidsdk/ui/LoginActivity.java

```diff
@@ -146,6 +146,9 @@ protected void onDestroy() {
 	@Override
     protected void onNewIntent(Intent intent) {
         super.onNewIntent(intent);
+
+        // Reloads login page for every new intent to ensure the correct login server is selected.
+        webviewHelper.loadLoginPage();
```

Comment:
Since this activity is a single task activity, `onCreate()` is triggered only once. Hence, we should update the login host here to the current value to ensure that we don't end up with a stale login host while coming in from the SP app. This can occur in some super corner cases without this fix.